src/rgw/rgw_sync_checkpoint.cc

```diff
@@ -49,8 +49,9 @@ bool operator<(const std::vector<rgw_bucket_shard_sync_info>& lhs,
 
 bool empty(const BucketIndexShardsManager& markers, int size)
 {
+  static const std::string empty_string;
   for (int i = 0; i < size; ++i) {
-    const auto& m = markers.get(i, "");
+    const auto& m = markers.get(i, empty_string);
```

Comment:
I don't think this is necessary? C++ String Literals have static storage duration and exist through the life of the program.

See notes in https://en.cppreference.com/w/cpp/language/string_literal

Reply:
but this is an std::string, not a string literal